### PR TITLE
Fix standalone role bootstrapping

### PR DESCRIPTION
**Pull request by ssbarnea**
- Fixes a bug introduce when we added support for project_dir and we still continued to rely on CWD in few places.
- Adds tests for covering this use-case

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -24,7 +24,7 @@ commands =
   ansible --version
   # We add coverage options but not making them mandatory as we do not want to force
   # pytest users to run coverage when they just want to run a single test with `pytest -k test`
-  coverage run -m pytest {posargs:}
+  coverage run -m pytest -vvvv {posargs:}
```
Comment (webknjaz) [inline tox.ini:27]: The level of verbosity should be contributed by the caller.

File: src/ansible_compat/runtime.py [hunk 2]
```diff
@@ -122,6 +122,7 @@ def exec(self, args: Union[str, List[str]]) -> CompletedProcess:
             check=False,
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
+            env=os.environ,
```
Comment (webknjaz) [inline src/ansible_compat/runtime.py:125]: This happens by default. Why did you add it?

File: pyproject.toml [hunk 3]
```diff
@@ -24,7 +24,8 @@ skip-string-normalization = true
 profile = "black"
 
 [tool.pytest.ini_options]
-addopts = "--no-success-flaky-report"
+addopts = "--no-success-flaky-report -vvvv"
```
Comment (webknjaz) [inline pyproject.toml:27]: This looks out of the scope also it's inconsistent with the comment below.

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[submodule "test/assets/ansible-role-mysql"]
```
Comment (webknjaz) [inline .gitmodules:1]: It'd go for git subtree or put the synching logic into the pytest setup. Making the contributors do this is fragile.

File: test/test_runtime.py [hunk 5]
```diff
@@ -87,6 +87,25 @@ def test_runtime_version_fail_cli(mocker: MockerFixture) -> None:
...
+    runtime = Runtime(isolated=True, project_dir=project_dir)
+    runtime.prepare_environment()
+    for record in caplog.records:
+        assert record.levelname != "CRITICAL"
+    assert "symlink to current repository in order to enable Ansible to find the role" in caplog.text
+    print(caplog.text)
+    # check that role appears as installed now
+    result = runtime.exec(["ansible-galaxy", "list", "-vvv"])
+    # print(result)
+    assert result.returncode == 0, result
```
Comment (webknjaz) [inline test/test_runtime.py:105]: No need to print out the `result` like this — just add `--showlocals` to addopts to reveal it.

**Comment by ssbarnea** [comment]
There is a lot of debugging junk i need to remove. I think I found the culprit yesterday, mainly submodules cloning no longer works with checkout v2. Instead I will just create a mini embedded role for testing.
